### PR TITLE
Fix warnings in Rssdp

### PR DESCRIPTION
**Pull request by Ullmie02**
Part of #2149

### CODE REVIEW

File: RSSDP/Infrastructure/DisposableManagedObjectBase.cs [hunk 1]
```diff
@@ -68,8 +69,7 @@ public string BuildMessage(string header, Dictionary<string, string> values)
         public void Dispose()
         {
             IsDisposed = true;
-
-            Dispose(true);
```
Comment (Bond-009) [inline RSSDP/Infrastructure/DisposableManagedObjectBase.cs:72]: Why?

File: RSSDP/Infrastructure/SsdpCommunicationsServer.cs [hunk 2]
```diff
@@ -32,46 +32,39 @@ public sealed class SsdpCommunicationsServer : DisposableManagedObjectBase, ISsd
...
+        public SsdpCommunicationsServer(
+            ISocketFactory socketFactory,
+            INetworkManager networkManager,
+            ILogger logger,
+            bool enableMultiSocketBinding)
             : this(socketFactory, 0, SsdpConstants.SsdpDefaultMulticastTimeToLive, networkManager, logger, enableMultiSocketBinding)
         {
-            
+            _broadcastListenSocketSynchroniser = new object();
+            _sendSocketSynchroniser = new object();
```
Comment (Bond-009) [inline RSSDP/Infrastructure/SsdpCommunicationsServer.cs:63]: Why?

File: RSSDP/Infrastructure/DisposableManagedObjectBase.cs (hunk 1 above)
Comment (Ullmie02) [reply RSSDP/Infrastructure/DisposableManagedObjectBase.cs:72]: I haven't read the warning correctly. Thought I have to replace `Dispose(true)` with `GC.SuppressFinalize(this)` but I think we want both

File: RSSDP/Infrastructure/SsdpCommunicationsServer.cs (hunk 2 above)
Comment (Ullmie02) [reply RSSDP/Infrastructure/SsdpCommunicationsServer.cs:63]: The fields were initialized at declaration and in one constructor. I moved the initialization to the constructors. But both constructors do the same so moving initialization back to declaration.

File: RSSDP/Infrastructure/ISsdpCommunicationsServer.cs [hunk 3]
```diff
@@ -39,14 +47,7 @@ public interface ISsdpCommunicationsServer : IDisposable
...
-        Task SendMulticastMessage(string message, int sendCount, IPAddress fromLocalIpAddress, CancellationToken cancellationToken);
 
-        /// <summary>
-        /// Gets or sets a boolean value indicating whether or not this instance is shared amongst multiple <see cref="SsdpDeviceLocatorBase"/> and/or <see cref="ISsdpDevicePublisher"/> instances.
-        /// </summary>
-        /// <remarks>
-        /// <para>If true, disposing an instance of a <see cref="SsdpDeviceLocatorBase"/>or a <see cref="ISsdpDevicePublisher"/> will not dispose this comms server instance. The calling code is responsible for managing the lifetime of the server.</para>
-        /// </remarks>
-        bool IsShared { get; set; }
+        Task SendMulticastMessage(string message, int sendCount, IPAddress fromLocalIpAddress, CancellationToken cancellationToken);
```
Comment (cvium) [inline RSSDP/Infrastructure/ISsdpCommunicationsServer.cs:51]: No XML doc?

**Pull request closed without merging**